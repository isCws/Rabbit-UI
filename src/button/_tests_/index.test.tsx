import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import Button from '../index';

describe('<Button />', () => {

    //测试输入的按钮文本是否正确
    test('should render button', () => {
        const { container } = render(
            <Button size='small' defaultProp={{}}>按钮一</Button>
        );
        expect(container).toMatchSnapshot();
    });

    //测试点击后的回调函数是否执行
    test('should touch callback', () => {
        const callback = jest.fn();
        const { getByText } = render(
            <Button
                size='small'
                defaultProp={{}}
                handleClick={callback}
            >
                按钮二
            </Button>
        )
        userEvent.click(getByText('按钮二'));
        expect(callback).toHaveBeenCalledTimes(0);
    })

    //测试按钮的禁选模块
    test('should be disabled', () => {
        const { container } = render(
            <Button
                size='small'
                defaultProp={{}}
                disabled={true}
            >
                按钮
            </Button>
        )
        expect(container).toBeDisabled;
    })
})