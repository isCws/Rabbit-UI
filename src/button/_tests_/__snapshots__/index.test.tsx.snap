// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button 1`] = `
<div>
  <button
    class="rabbit-btn btn-small primary"
    htmltype="button"
  >
    <div />
    按钮一
  </button>
</div>
`;
